source/common/network/socket_impl.h

```diff
@@ -126,11 +126,11 @@ class SocketImpl : public virtual Socket {
   IoHandle& ioHandle() override { return *io_handle_; }
   const IoHandle& ioHandle() const override { return *io_handle_; }
   void close() override {
-    if (io_handle_->isOpen()) {
+    if (io_handle_ && io_handle_->isOpen()) {
```

Comment:
I'm surprised to see this new null check here. In a previous comment, there is some mention of allowing null sockets for tests. Is that what this is about? If so, what are the circumstances where we end up with null sockets?

Reply:
so all of the socket -> iohandle swaps are because if we fail socket creation we end up with a socket class but no iohandle class.  Which unfortunately means my "allow for failed tests" release assert above isn't WAI because it'll fail unit tests and fail if we don't create the socket.  I'm going to flip this off for now because I don't think I have time to fix the unit tests such that I can do a pure isOpen check.